system-parachains/coretime/coretime-polkadot/src/lib.rs

```diff
@@ -0,0 +1,1102 @@
+#[cfg(feature = "std")]
+use sp_version::NativeVersion;
+use sp_version::RuntimeVersion;
+use system_parachains_constants::{
+	polkadot::{consensus::*, currency::*, fee::WeightToFee},
+	AVERAGE_ON_INITIALIZE_RATIO, HOURS, MAXIMUM_BLOCK_WEIGHT, NORMAL_DISPATCH_RATIO, SLOT_DURATION,
+};
+use weights::{BlockExecutionWeight, ExtrinsicBaseWeight, RocksDbWeight};
+use xcm::prelude::*;
+use xcm_config::{
+	DotRelayLocation, FellowshipLocation, GovernanceLocation, StakingPot,
+	XcmOriginToTransactDispatchOrigin,
+};
+use xcm_runtime_apis::{
+	dry_run::{CallDryRunEffects, Error as XcmDryRunApiError, XcmDryRunEffects},
+	fees::Error as XcmPaymentApiError,
+};
+
+/// The address format for describing accounts.
+pub type Address = MultiAddress<AccountId, ()>;
+
+/// Block type as expected by this runtime.
+pub type Block = generic::Block<Header, UncheckedExtrinsic>;
+
+/// A Block signed with a Justification
+pub type SignedBlock = generic::SignedBlock<Block>;
+
+/// BlockId type as expected by this runtime.
+pub type BlockId = generic::BlockId<Block>;
+
+/// The SignedExtension to the basic transaction logic.
+pub type SignedExtra = (
+	frame_system::CheckNonZeroSender<Runtime>,
+	frame_system::CheckSpecVersion<Runtime>,
+	frame_system::CheckTxVersion<Runtime>,
+	frame_system::CheckGenesis<Runtime>,
+	frame_system::CheckEra<Runtime>,
+	frame_system::CheckNonce<Runtime>,
+	frame_system::CheckWeight<Runtime>,
+	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
+	frame_metadata_hash_extension::CheckMetadataHash<Runtime>,
+	cumulus_primitives_storage_weight_reclaim::StorageWeightReclaim<Runtime>,
+);
+
+/// Unchecked extrinsic type as expected by this runtime.
+pub type UncheckedExtrinsic =
+	generic::UncheckedExtrinsic<Address, RuntimeCall, Signature, SignedExtra>;
+
+/// Migrations to apply on runtime upgrade.
+pub type Migrations = (pallet_xcm::migration::MigrateToLatestXcmVersion<Runtime>,);
+
+/// Executive: handles dispatch to the various modules.
+pub type Executive = frame_executive::Executive<
+	Runtime,
+	Block,
+	frame_system::ChainContext<Runtime>,
+	Runtime,
+	AllPalletsWithSystem,
+	Migrations,
+>;
+
+impl_opaque_keys! {
+	pub struct SessionKeys {
+		pub aura: Aura,
+	}
+}
+
+#[sp_version::runtime_version]
+pub const VERSION: RuntimeVersion = RuntimeVersion {
+	spec_name: create_runtime_str!("coretime-polkadot"),
+	impl_name: create_runtime_str!("coretime-polkadot"),
+	authoring_version: 1,
+	spec_version: 1_003_000,
+	impl_version: 0,
+	apis: RUNTIME_API_VERSIONS,
+	transaction_version: 0,
+	state_version: 1,
+};
+
+/// The version information used to identify this runtime when compiled natively.
+#[cfg(feature = "std")]
+pub fn native_version() -> NativeVersion {
+	NativeVersion { runtime_version: VERSION, can_author_with: Default::default() }
+}
+
+parameter_types! {
+	pub const Version: RuntimeVersion = VERSION;
+	pub RuntimeBlockLength: BlockLength =
+		BlockLength::max_with_normal_ratio(5 * 1024 * 1024, NORMAL_DISPATCH_RATIO);
```

Comment:
Is block length same as PoV size?

Reply:
The block length here is the limit set for the total length of each extrinsic class (normal/operational/mandatory). Here we explicitly set it to the same values as `BlockLength::default()` - the first argument sets the max for operational and mandatory, the normal is a `Perbill` which is multiplied by the max, in this case Normal extrinsics can only take up 75% of the block.
I think of it as the PoV limit and respective ratio, but that's maybe an oversimplification.

Reply:
I am asking, because if this is really only the block (without state proof), then this is actually too large. Total PoV size is limited to 5Meg (compressed, but still).

Reply:
I think it is fine and I think this is that 5 Meg limit, this is what we have set for all system chains, and is also the default value.
But maybe @skunert could help here to confirm that it is correctly set

Reply:
Yes that should be the correct setting. It does not matter so much however, since the `BlockLength` is mostly legacy. It will never reach the 5MB limit because `BlockWeight` tracks proof size and length combined. So it will hit the limit long before `BlockLength`.